src/server/decode.rs

```diff
@@ -16,6 +16,13 @@ const LF: u8 = b'\n';
 /// The number returned from httparse when the request is HTTP 1.1
 const HTTP_1_1_VERSION: u8 = 1;
 
+#[derive(Debug, Clone)]
+pub(crate) enum BodyType {
```

Comment:
Should we keep this to `u32` rather than 2 `usize` since browsers only supports 2GB content length? And 
```rust
pub(crate) enum BodyType {
    FixedLength(NonZeroU32),
    Chunked,
}
```

Reply:
I don't think the assumption that the client is a mainstream browser should be baked into the library.

Reply:
Okay, but what if we make it `NonZeroUsize`? I think it is quite wasteful to use one usize for only one bit.

Reply:
There are three variants anyway. Even if there weren't it corresponds to the Content-Length header which can be 0.

Reply:
Oh, I didn't know Content-Length can be zero. But I don't understand why `BodyType` have `Close`, if it is closed means there are no body right?

Reply:
no, it means the end of the body is indicated by the connection being closed. in http terms if the content length isn't specified and the transfer encoding isn't chunked.

no body is a separate concern, either the connection is closed without sending more data, content-length is set to 0 or transfer encoding is chunked and a 0 size chunk is sent